### PR TITLE
Mining Rework aka Cursed Lavaland

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

This pull request ports and creates a large amount of mining features

Something strange happened to github, so I can't link initial PRs at all, they no longer exitst(?) apparently.
A large portion of concepts and code was ported from https://github.com/Skyrat-SS13/Skyrat13
Initial PRs by BobJoga

## Why It's Good For The Game

New bosses are good.

## Changelog
:cl:
add: Added a new mining Z level, called Abyss! It has it's own fauna and bosses, as well as some tasty loot from them!
add: Falling down the chasms now drops you to the Abyss. You can get out there by killing Gladiator and The Legion
tweak: Goliath Cloak now is a propper hooded cloak. However, it was nerfed so it won't be OP.
add: SEVA and EXO suits! SEVA is a fragile, but storm-proof and allows goliaths to stun you for longer, while EXO is a reinforced version of explorer suit, which shortents goliath stuns, but has a decent slowdown.
add: Premium KC. Right now it's only obtainable through admin shenanigans or as a rare drop from Crazy Miners
add: New KC trophies. For example, Legion's(Huge one) trophy has a 15% chance of spawning a friendly skull on mark detonation
add: Abyssal energy, energy canisters and power collector ruins. In those you can get a canister full of energy and later splash it on some boss(Drake, Bubblegum or Colossus) to enrage it and get better loot without trying to get the boss into the Abyss for that.
add: Suit vouchers for miners! This is the only way you can obtain EXO and SEVA suits 
add: Parachutes that will allow you to drop into chasms without any damage
add: Megafauna KA modkits! Customise your KA in any way you like
add: Harder "Enraged" versions of Colossus, Ash Drake and Bubblegum! They give much better loot, however, they are very deadly.
add: Corrupted System boss. Its a broken mining mech with an experimental AI, which now seeks for blood. Kill it for a new plasma cutter and combat mecha drill
add: The Gladiator! An immortal ash walker boss, who guards the Abyssal part of Legion's dungeon.
add: Abyssal Demons, Fire Wisps and Ash Whelps as new Abyss mobs!
add: Crazy Miners, who will hunt local fauna and try to kill it. They also can use medipens, healing themselves up to 3 times.
add: Premium KA! (Right now non-obtainable for balance reasons)
add: Luxury medipens with 107u of reagents! Can be obtained in mining vendors for 1700 points
server: Loading maps now takes 10 more seconds.
/:cl:

### CODE REVIEW

**Comment by PowerfulBacon** [comment]
Provide or links where you got them from anyway

**Comment by PowerfulBacon** [comment]
Looking at this, I really dont like the idea of death squad armour as a drop (even with 5 less damage resist), that stuff is made to be ridiculously overpowered and have no chance of stopping. Also a lot of the armour values on the suits seem pretty high.

**Comment by PowerfulBacon** [comment]
Looking at this, I really dont like the idea of death squad armour as a drop (even with 5 less damage resist), that stuff is made to be ridiculously overpowered and have no chance of stopping. Also a lot of the armour values on the suits seem pretty high.

**Comment by SmArtKar** [comment]
Check HECK armor values. But yeah, gonna nerf it a bit later. (Its really fucking hard to get)

**Comment by SmArtKar** [comment]
> Provide or links where you got them from anyway

BobJoga's account got deleted with all PRs.

**Comment by PowerfulBacon** [comment]
I wouldnt be particularly against other mining items getting nerfed values when used on station (or in high pressure)

**Comment by SmArtKar** [comment]
> I wouldnt be particularly against other mining items getting nerfed values when used on station (or in high pressure)

image

Upper - HECK. Lower - Praetorian one. Killing Enraged Bubblegum is REALLY hard even with top-tier gear

**Comment by AnjelicaChaykovsky** [comment]
It doesn't have the Great wolf sif boss ? Sad doggo moment.
Last time I saw him in game there were bunch of corpses with heck suit on them near him

**Comment by SmArtKar** [comment]
> It doesn't have the Great wolf sif boss ? Sad doggo moment.
> Last time I saw him in game there were bunch of corpses with heck suit on them near him

hmmm, maybe later will add him. and not just Great wolf, he is The Great Brown Wolf Sif

**Comment by St0rmC4st3r** [comment]
I don't like this. It is my personal opinion, but it is the opposite direction of what I would like mining to be.
Miners already have an idiotic mortality rate and they never actually mine beyond minimum.

**Comment by SmArtKar** [comment]
> I don't like this. It is my personal opinion, but it is the opposite direction of what I would like mining to be.
> Miners already have an idiotic mortality rate and they never actually mine beyond minimum.

We have BS miners which make ore mining unneeded.

**Comment by VictorVow** [comment]
> We have BS miners which make ore mining unneeded.

You're saying this like this is the direction mining should go. Do you not even read the job title? **Mining.** Not something that should be done by magic material generators, which should be removed if it's being used as a justification for further deprecation of ore mining rather than as a supplement to it.

From seeing Prae suit and other powercreep, it's clear as ever for me that the only way mining can work long-term is if Lavaland combat instead relies on mechs which only work there on that Z-level.

**Comment by SmArtKar** [comment]
> > We have BS miners which make ore mining unneeded.
> 
> You're saying this like this is the direction mining should go. Do you not even read the job title? **Mining.** Not something that should be done by magic material generators, which should be removed if it's being used as a justification for further deprecation of ore mining rather than as a supplement to it.
> 
> From seeing Prae suit and other powercreep, it's clear as ever for me that the only way mining can work long-term is if Lavaland combat instead relies on mechs which only work there on that Z-level.

Did you EVER played mining? Like really, not joking. And heard of H.E.C.K. suit? Yeah, almost the same

**Comment by SomeAngryMiner** [comment]
while concept is relatively good, i really, *really* dont like adding even more gamer gear to miners, and incentives to just dropkick yourself down a shaft, not mine a thing, and return with (even more) deathsquad level equipment (even then HECK has the illusion drawback, that when hits you it briefly stuns you) to validhunt the funny man with esword.

**Comment by Hibabo** [comment]
I think its fine, I haven't fought against any of the bosses in the abyss, but usually a good miner can balance between getting enough ores to station and killing fauna.

It might also take a hot minute to get stuff done down in the abyss which means by the time the miner gets to the station and is ready to socialize the round may already be over, but I can't say without actually trying it out, I'm all for more lavaland content though since gateway was stripped of its glory.

edit: a bunch of it is broken english though.

**Comment by SomeAngryMiner** [comment]
as long as its balanced, looks good, yeah. 
Gotta fix those 18 travis errors first though.

**Comment by SmArtKar** [comment]
> while concept is relatively good, i really, _really_ dont like adding even more gamer gear to miners, and incentives to just dropkick yourself down a shaft, not mine a thing, and return with (even more) deathsquad level equipment (even then HECK has the illusion drawback, that when hits you it briefly stuns you) to validhunt the funny man with esword.

Here's my point, if you can kill Enraged Bubblegum, then you don't even need his gear to kill that esword ERPer

**Comment by eeSPee** [comment]
I think here are the key concerns;
* draws focus away from mining ores and incentives miners to hunt mobs
* provides OP gear
* other departments rely on mining to work, but mining doesn't need much help from other departments

**Comment by Moccha-Bee** [comment]
If you want miners to get minerals just like, make them not die ????? Every time i hear some out of touch scinece main saying that "miners always go to kill mobs" is completly out of touch to what really happens to miners, most of them, and i'm not kidding , die to megafauna aggroing them from 10 tiles away or getting hit by goliath tentacles, they din't want to fight them they died because they are unrobust, removing fauna from lavaland or trying to make their drops very limited is just going to kill any incetive for mining, no one likes shooting at rocks for one hour straight just to get a better rock cutter and so better ore detection stuff. The best miners are the ones who go to kill megafauna as they always get the gamer loot and in return can mining more efficiently and not die.

**Comment by utilbird** [comment]
More miner powercreep 😳 

When are we removing all lavaland content except for the pickaxe

**Comment by Ratvarr** [comment]
prehaps make a new role called "lavaland "helper" as a role in cargo that has a bit more shit meant for killing but then miner does not

File: code/datums/components/chasm.dm [hunk 1]
```diff
@@ -100,8 +100,16 @@
...
+			if(ishuman(L))
+				var/mob/living/carbon/human/H = L
+				if(istype(H.get_item_by_slot(ITEM_SLOT_BACK), /obj/item/parachute))
+					var/obj/item/parachute/para = H.get_item_by_slot(ITEM_SLOT_BACK)
+					para.deploy(T)
+					falling_atoms -= AM
+					return
+
+			L.Knockdown(100)
+			L.adjustBruteLoss(55) //Buffed it, was too weak
```
Comment (zeskorion) [inline code/datums/components/chasm.dm:112]: why are you buffing this? it's designed for multiz shit, not just lavaland

File: code/datums/ruins/lavaland.dm [hunk 2]
```diff
@@ -128,16 +146,15 @@
 	id = "hierophant"
 	description = "A strange, square chunk of metal of massive size. Inside awaits only death and many, many squares."
 	suffix = "lavaland_surface_hierophant.dmm"
-	always_place = TRUE
-	allow_duplicates = FALSE
+	always_place = TRUE //Now multiple can spawn
```
Comment (zeskorion) [inline code/datums/ruins/lavaland.dm:149]: no, there do not need to be multiple of these at any given time

File: code/game/mecha/equipment/tools/mining_tools.dm [hunk 3]
```diff
@@ -135,6 +135,17 @@
...
 
+/obj/item/mecha_parts/mecha_equipment/drill/experimental
+	name = "experimental exosuit drill"
+	desc = "An experimental exosuit drill, equipped with hydraulic system and made in light-green colors."
+	icon = 'icons/obj/lavaland/artefacts.dmi'
+	icon_state = "corrupted_drill"
+	equip_cooldown = 5
+	drill_delay = 2
+	drill_level = DRILL_HARDENED
+	force = 30
```
Comment (zeskorion) [inline code/game/mecha/equipment/tools/mining_tools.dm:146]: you realize this is the force from hitting someone with the item itself, correct?

File: code/game/turfs/open/floor/plating/asteroid.dm [hunk 4]
```diff
@@ -151,10 +152,10 @@
...
 
 /turf/open/floor/plating/asteroid/airless/cave/volcanic
-	mob_spawn_list = list(/mob/living/simple_animal/hostile/asteroid/goliath/beast/random = 50, /obj/structure/spawner/lavaland/goliath = 3, \
-		/mob/living/simple_animal/hostile/asteroid/basilisk/watcher/random = 40, /obj/structure/spawner/lavaland = 2, \
-		/mob/living/simple_animal/hostile/asteroid/hivelord/legion/random = 30, /obj/structure/spawner/lavaland/legion = 3, \
-		SPAWN_MEGAFAUNA = 6, /mob/living/simple_animal/hostile/asteroid/goldgrub = 10, )
+	mob_spawn_list = list(/mob/living/simple_animal/hostile/asteroid/goliath/beast/random = 50, /obj/structure/spawner/lavaland/goliath = 4, \
+		/mob/living/simple_animal/hostile/asteroid/basilisk/watcher/random = 30, /obj/structure/spawner/lavaland = 2, \
+		/mob/living/simple_animal/hostile/asteroid/hivelord/legion/random = 30, /obj/structure/spawner/lavaland/legion = 4, \
+		SPAWN_MEGAFAUNA = 6, /mob/living/simple_animal/hostile/asteroid/goldgrub = 10, /mob/living/simple_animal/hostile/asteroid/crazy_miner/random = 10)
```
Comment (zeskorion) [inline code/game/turfs/open/floor/plating/asteroid.dm:158]: why are you spawning these mobs? they can drop sonic jackhammers quite early in the round

File: code/modules/clothing/suits/cloaks.dm [hunk 5]
```diff
@@ -55,16 +55,19 @@
...
 	desc = "A staunch, practical cape made out of numerous monster materials, it is coveted amongst exiles & hermits."
 	allowed = list(/obj/item/flashlight, /obj/item/tank/internals, /obj/item/pickaxe, /obj/item/twohanded/spear, /obj/item/twohanded/bonespear, /obj/item/organ/regenerative_core/legion, /obj/item/kitchen/knife/combat/bone, /obj/item/kitchen/knife/combat/survival)
-	armor = list("melee" = 50, "bullet" = 10, "laser" = 25, "energy" = 10, "bomb" = 25, "bio" = 0, "rad" = 0, "fire" = 60, "acid" = 60) //a fair alternative to bone armor, requiring alternative materials and gaining a suit slot
+	armor = list(MELEE = 15, BULLET = 10, LASER = 10, ENERGY = 10, BOMB = 5, BIO = 0, RAD = 0, FIRE = 40, ACID = 30) //a fair alternative to bone armor, requiring alternative materials and gaining a suit slot
 	hoodtype = /obj/item/clothing/head/hooded/cloakhood/goliath
 	body_parts_covered = CHEST|GROIN|ARMS
 	resistance_flags = FIRE_PROOF
 
+	slot_flags = ITEM_SLOT_NECK
+	slot_type = ITEM_SLOT_NECK
```
Comment (zeskorion) [inline code/modules/clothing/suits/cloaks.dm:64]: 15 armor on the neck? fuck no, armor stacking is already good enough

File: code/modules/mining/equipment/explorer_gear.dm [hunk 6]
```diff
@@ -114,3 +114,75 @@
...
+
+/obj/item/clothing/mask/gas/exo
+	name = "Exosuit Mask"
+	desc = "A face-covering mask that can be connected to an air supply. Intended for use with the Exosuit."
+	icon_state = "exo"
+	resistance_flags = FIRE_PROOF
+
+/**************** Praetor Suit. Exclusive super bubblegum loot ****************/
+
+/obj/item/clothing/suit/space/hardsuit/deathsquad/praetor
```
Comment (zeskorion) [inline code/modules/mining/equipment/explorer_gear.dm:172]: this reference is waaaaay too fucking overt. change that up a good fucking bit

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 7]
```diff
@@ -29,6 +29,24 @@
 	var/backstab_bonus = 30
 	var/light_on = FALSE
 	var/brightness_on = 5
+	var/destabilizer_type = /obj/item/projectile/destabilizer
+
+/obj/item/twohanded/kinetic_crusher/premium
```
Comment (zeskorion) [inline code/modules/mining/equipment/kinetic_crusher.dm:34]: crushers are already the meta. Why are you buffing that meta?

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 8]
```diff
@@ -443,3 +461,117 @@
...
+
+/obj/item/crusher_trophy/abyssal_crystal
+	name = "abyssal crystal"
+	desc = "A red, glowing crystal that was sliced off the Abyssal Demon. Suitable as a trophy for a kinetic crusher"
+	icon_state = "crystal_shard"
+	denied_type = /obj/item/crusher_trophy/abyssal_crystal
+	bonus_value = 5
+
+/obj/item/crusher_trophy/abyssal_crystal/effect_desc()
+	return "mark detonation to pierce target with crystal shards and stun him for [bonus_value * 0.1] seconds"
```
Comment (zeskorion) [inline code/modules/mining/equipment/kinetic_crusher.dm:494]: him?

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 9]
```diff
@@ -443,3 +461,117 @@
...
+/obj/effect/temp_visual/abyss_shards
+	name = "abyssal shards"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "bloody_trap"
+	duration = 10
+	layer = BELOW_MOB_LAYER
+
+/obj/item/crusher_trophy/abyssal_crystal/on_mark_detonation(mob/living/target, mob/living/user)
+	new /obj/effect/temp_visual/abyss_shards(get_turf(target))
+	target.Stun(bonus_value * 0.1)
```
Comment (zeskorion) [inline code/modules/mining/equipment/kinetic_crusher.dm:505]: this won't do anything to anything but carbons, who marks cant be detonated on- unless, god forbid, you changed that fact

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 10]
```diff
@@ -443,3 +461,117 @@
...
+	target.adjustBruteLoss(rand(5,10))
+
+//Fire Wisp
+
+/obj/item/crusher_trophy/tail_spike/firestone
+	name = "firestone"
+	desc = "A stone, literraly made out of fire. Suitable as a trophy for a kinetic crusher"
```
Comment (zeskorion) [inline code/modules/mining/equipment/kinetic_crusher.dm:512]: misspelled and dumb description. doesnt even have a use description

File: code/modules/mining/equipment/kinetic_crusher.dm [hunk 11]
```diff
@@ -443,3 +461,117 @@
...
+	name = "ash whelp's wing"
+	desc = "An ash whelp's wing, carefully sliced off. Suitable as a trophy for a kinetic crusher"
+	icon_state = "crystal_shard"
+	bonus_value = 5
+
+//Corrupted System
+
+/obj/item/crusher_trophy/rogue_ai
+	name = "rogue ai"
+	desc = "A potato with an AI incide it. Not surprisingly that it got corrupted. Suitable as a trophy for a kinetic crusher"
```
Comment (zeskorion) [inline code/modules/mining/equipment/kinetic_crusher.dm:531]: SPELLING!

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 12]
```diff
@@ -83,6 +83,67 @@
...
+/obj/item/clothing/neck/necklace/golden_amulet/proc/unequip(var/mob/user)
+	UnregisterSignal(user, COMSIG_MOB_DEATH)
+
+/obj/item/clothing/neck/necklace/golden_amulet/proc/revival()
+	if(used)
+		return
+	var/mob/living/user = loc
+	if(!istype(user))
+		return
+	user.revive(full_heal = 1, admin_revive = 1)
```
Comment (zeskorion) [inline code/modules/mining/lavaland/necropolis_chests.dm:124]: NO. GOD, PLEASE NO.

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 13]
```diff
@@ -83,6 +83,67 @@
...
+/obj/item/throwing_belt
+	name = "ancient belt"
+	desc = "A red belt, made out of sinew. It is full of throwing daggers!"
+	icon = 'icons/obj/clothing/belts.dmi'
+	icon_state = "ebelt"
+	lefthand_file = 'icons/mob/inhands/equipment/belt_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/belt_righthand.dmi'
+	slot_flags = ITEM_SLOT_BELT
+
+/obj/item/throwing_belt/attack_hand(var/mob/living/carbon/human/H)
```
Comment (zeskorion) [inline code/modules/mining/lavaland/necropolis_chests.dm:138]: how are you supposed to easily pick this up?

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 14]
```diff
@@ -906,6 +992,43 @@
...
+/obj/item/melee/ghost_sword/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	var/ghost_counter = ghost_check()
+	final_block_chance += CLAMP((ghost_counter * 15), 0, 75)
+	owner.visible_message("<span class='danger'>[owner] is protected by a ring of [ghost_counter] ghosts!</span>")
+	return ..()
+
+/obj/item/dragons_blood/pure
+	name = "bottle of pure dragons blood"
+
+/obj/item/dragons_blood/pure/attack_self(mob/living/carbon/human/user)
```
Comment (zeskorion) [inline code/modules/mining/lavaland/necropolis_chests.dm:1016]: furry content, also should add wings here

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 15]
```diff
@@ -1124,13 +1265,94 @@
...
+		ADD_TRAIT(src, TRAIT_NODROP, MEGAFAUNA_TRAIT)
+		armor = initial(armor)
+	else
+		armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0)
+
+/obj/item/fallen_wings/dropped(var/mob/user)
+	. = ..()
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0)
+
+/obj/item/fallen_wings/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
```
Comment (zeskorion) [inline code/modules/mining/lavaland/necropolis_chests.dm:1323]: doesn't work with our block refactor

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 16]
```diff
@@ -1413,3 +1635,127 @@
 			new /obj/item/wisp_lantern(src)
 		if(3)
 			new /obj/item/prisoncube(src)
+
+/obj/item/twohanded/crucible
```
Comment (zeskorion) [inline code/modules/mining/lavaland/necropolis_chests.dm:1639]: same change as praetor suit- waaay too overt here

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 17]
```diff
@@ -1413,3 +1635,127 @@
...
+	hitsound = "swing_hit"
+	var/hitsound_on = 'sound/weapons/bladeslice.ogg'
+	armour_penetration = 50
+	light_color = "#ff0000"//BLOOD RED
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	block_power = 50
+	block_level = 1
+	block_upgrade_walk = 1
+	block_flags = BLOCKING_ACTIVE | BLOCKING_NASTY | BLOCKING_PROJECTILE
+	max_integrity = 400
```
Comment (zeskorion) [inline code/modules/mining/lavaland/necropolis_chests.dm:1664]: why does this have an integrity value?

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 18]
```diff
@@ -1413,3 +1635,127 @@
...
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 100, "acid" = 100, "wound" = 35)
```
Comment (zeskorion) [inline code/modules/mining/lavaland/necropolis_chests.dm:1665]: why does this have wound armor? we dont have wounds, and objects cant get wounds

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 19]
```diff
@@ -1413,3 +1635,127 @@
...
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	var/brightness_on = 6
+	var/item_state_on = "crucible1"
+
+/obj/item/twohanded/crucible/suicide_act(mob/living/carbon/user)
```
Comment (zeskorion) [inline code/modules/mining/lavaland/necropolis_chests.dm:1670]: I'm not even going to bother reviewing this file from this point down, i already have a headache and i dont wanna make it worse

File: code/modules/mining/parachute_and_ropes.dm [hunk 20]
```diff
@@ -0,0 +1,49 @@
+/obj/item/parachute
```
Comment (zeskorion) [inline code/modules/mining/parachute_and_ropes.dm:1]: you are aware this is useable on all multiz? it should have a description denoting it as such

File: code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm [hunk 21]
```diff
@@ -69,6 +69,10 @@ Difficulty: Hard
 							   /datum/action/innate/megafauna_attack/blood_warp)
 	small_sprite_type = /datum/action/small_sprite/megafauna/bubblegum
 
+	abyss_born = TRUE
+	enraged_type = /mob/living/simple_animal/hostile/megafauna/bubblegum/hard
+	enrage_message = "starts growing bigger as it feels the energy of Abyss flowing in his veins!"
```
Comment (zeskorion) [inline code/modules/mob/living/simple_animal/hostile/megafauna/bubblegum.dm:74]: this message hurts me inside

**Review by zeskorion — CHANGES_REQUESTED** [review]
I'm tired. I doubt this will ever be merged, looking at both the code and balance. 
Bobjoga is incredibly terrible at balancing stuff, and his code is... questionable, at best. You also failed to properly attribute this port according to code guidelines.

if no other maintainers object, I'm going to close this tommorow

File: code/game/turfs/open/floor/plating/asteroid.dm (hunk 4 above)
Comment (SmArtKar) [reply code/game/turfs/open/floor/plating/asteroid.dm:158]: That's not jackhammer ones... Thats new mob which is basically an NPC miner(still hostile to miners)

File: code/modules/mining/equipment/kinetic_crusher.dm (hunk 7 above)
Comment (SmArtKar) [reply code/modules/mining/equipment/kinetic_crusher.dm:34]: It has really small chance to drop

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 12 above)
Comment (SmArtKar) [reply code/modules/mining/lavaland/necropolis_chests.dm:124]: ITS A ONE USE FUCKING SUPERDRAKE DROP WHAT DO YOU WANT

File: code/modules/mining/equipment/kinetic_crusher.dm (hunk 9 above)
Comment (SmArtKar) [reply code/modules/mining/equipment/kinetic_crusher.dm:505]: 1. You can't detonate marks on humans
2. You CAN stun animals.
/mob/living/proc/AdjustStun(amount, updating = TRUE, ignore_canstun = FALSE) //Adds to remaining duration

File: code/modules/mining/parachute_and_ropes.dm (hunk 20 above)
Comment (SmArtKar) [reply code/modules/mining/parachute_and_ropes.dm:1]: yep

File: code/modules/mining/equipment/kinetic_crusher.dm (hunk 7 above)
Comment (VictorVow) [reply code/modules/mining/equipment/kinetic_crusher.dm:34]: If it's powerful then make it rare is why ninja is such shit balance as well. It's a lazy excuse for balancing just hiding something behind RNG. At least PerishedFraud's lavaland PRs are trying to fix the dumb RNG, this takes it the opposite direction.

**Comment by zeskorion** [comment]
This is
* poorly balanced.
* terrible shitcode, which you obviously lack the competence to properly fix (honestly i dont think most could fix this)
* an insane buff to what is already one of the game's most poorly designed jobs
* fails to follow contribution guidelines for ports

I'm closing this. Please do not attempt to port this content again- it needs to be coded from scratch in a more sensible and balanced way

**Pull request closed without merging**